kong/runloop/handler.lua

```diff
@@ -1134,6 +1108,25 @@ return {
   },
   access = {
     before = function(ctx)
+      if kong.configuration.log_level == "debug" then
+        -- If this is a version-conditional request, abort it if this dataplane has not processed at least the
+        -- specified configuration version yet.
+        local if_kong_transaction_id = kong.request and kong.request.get_header('x-if-kong-test-transaction-id')
```

Comment:
style nit, just double quotes:
```suggestion
        local if_kong_transaction_id = kong.request and kong.request.get_header("X-If-Kong-Test-Transaction-ID")
```

I am not fan of having `Test` in `X-If-Kong-Rest-Transaction-ID`, also the `X-` is kinda deprecated:
https://datatracker.ietf.org/doc/html/rfc6648

I know kong uses it in many places, but perhaps for new stuff we should not use it and just have:
`If-Kong-Test-Transaction-ID` or `If-Kong-Transaction-ID`.

Reply:
We should keep the `-Test` in the header name to signal this is a test only feature, it is a specific request from myself and @hbagdi. Please fix it.